app/src/main/java/com/example/concordia_campus_guide/Models/Helpers/CalendarViewModel.java

```diff
@@ -0,0 +1,110 @@
+package com.example.concordia_campus_guide.Models.Helpers;
+
+import android.Manifest;
+import android.content.ContentUris;
+import android.content.Context;
+import android.content.pm.PackageManager;
+import android.database.Cursor;
+import android.net.Uri;
+import android.provider.CalendarContract;
+import android.provider.CalendarContract.Instances;
+
+import androidx.core.app.ActivityCompat;
+import androidx.core.content.ContextCompat;
+import androidx.lifecycle.ViewModel;
+
+import com.example.concordia_campus_guide.Activities.SearchActivity;
+import com.example.concordia_campus_guide.Models.CalendarEvent;
+
+import java.util.Date;
+import java.util.concurrent.TimeUnit;
+
+public class CalendarViewModel extends ViewModel {
+
+    // The indices for the projection array above.
+    private static final int PROJECTION_TITLE_INDEX = 0;
+    private static final int PROJECTION_LOCATION_INDEX = 1;
+    private static final int PROJECTION_START_INDEX = 2;
+
+    public static final String[] INSTANCE_PROJECTION = new String[] {
+            Instances.TITLE,
+            Instances.EVENT_LOCATION,
+            Instances.DTSTART
+    };
+
+    public CalendarEvent getEvent(Context context, SearchActivity searchActivity) {
```

Comment:
Since we are using context for many methods in this class, it might make sense to make a Context attribute in this class, which we instantiate in the constructor. This could be done by extending the AndroidViewModel class instead. Then we could rewrite all methods that require the Context as a parameter to access the context through the class attribute

Reply:
That's a good point, I will do that.